gtsfm/loader/loader_base.py

```diff
@@ -196,25 +219,42 @@ def get_camera_intrinsics(self, index: int) -> Optional[gtsfm_types.CALIBRATION_
         if intrinsics_full_res is None:
             raise ValueError(f"No intrinsics found for index {index}.")
 
-        if intrinsics_full_res.fx() <= 0:
-            raise RuntimeError("Focal length must be positive.")
+        return self.__rescale_intrinsics(intrinsics_full_res, index)
 
-        if intrinsics_full_res.px() <= 0 or intrinsics_full_res.py() <= 0:
-            raise RuntimeError("Principal point must have positive coordinates.")
+    def get_gt_camera_intrinsics_full_res(self, index: int) -> Optional[gtsfm_types.CALIBRATION_TYPE]:
+        """Get the GT camera intrinsics at the given index, valid for a full-resolution image.
 
-        img_full_res = self.get_image_full_res(index)
-        # no downsampling may be required, in which case scale_u and scale_v will be 1.0
-        scale_u, scale_v, _, _ = img_utils.get_downsampling_factor_per_axis(
-            img_full_res.height, img_full_res.width, self._max_resolution
-        )
-        rescaled_intrinsics = Cal3Bundler(
-            fx=intrinsics_full_res.fx() * scale_u,
-            k1=0.0,
-            k2=0.0,
-            u0=intrinsics_full_res.px() * scale_u,
-            v0=intrinsics_full_res.py() * scale_v,
-        )
-        return rescaled_intrinsics
+        By default, this is implemented to return the same value as `get_camera_intrinsics_full_res`. However, this can
+        be overridden by subclasses to return a superior ground truth intrinsics.
+
+        Args:
+            the index to fetch.
+
+        Returns:
+            intrinsics for the given camera.
+        """
+        return self.get_camera_intrinsics_full_res(index)
+
+    def get_gt_camera_intrinsics(self, index: int) -> Optional[gtsfm_types.CALIBRATION_TYPE]:
+        """Get the GT camera intrinsics at the given index, for a possibly resized image.
+
+        Determine how the camera intrinsics and images should be jointly rescaled based on desired img. resolution.
+        Each loader implementation should set a `_max_resolution` attribute.
+
+        The GT camera intrinsics are the same as the camera intrinsics by default, but can be overridden by defining
```

Comment:
this sentence is a little weird: "The GT camera intrinsics are the same as the camera intrinsics by default,". maybe we mean "...are the same as the camera instrinsics returned by the loader in general for certain loader implementations, such as the COLMAP loader?"

Reply:
"default" was about the base class implementation, I reworded a bit